test_util/aws.py

```diff
@@ -170,30 +195,14 @@ def empty_and_delete_s3_bucket_from_stack(self):
         bucket.delete()
         log.info('Delete successfully triggered for {}'.format(self.stack.stack_name))
 
-    def get_master_ips(self, state_list=None):
-        instances = self.get_group_instances(['MasterServerGroup'], state_list)
-        return instances_to_hosts(instances)
-
-    def get_public_agent_ips(self, state_list=None):
-        instances = self.get_group_instances(['PublicSlaveServerGroup'], state_list)
-        return instances_to_hosts(instances)
+    def get_master_ips(self):
+        return instances_to_hosts(self.get_auto_scaling_instances('MasterServerGroup'))
```

Comment:
re: caching how would we ever deal with nodes that are terminated or added? Perhaps check to see if any stack new stack events have occurred. If not, we don't need to worry about refreshing

The running filter is tricky because last time I tried that I was getting only a subset of the expected nodes after the cloudformation reported "CREATE_COMPLETE" because apparently the labeling of the instances as 'running' is sometimes a little bit delayed. That was an old implementation so I can try it again to prove whether or not this is really a problem.